syntax = "proto3";

package rpc;

message empty
{}

message root
{
  string value = 1;
}

message add_root_result
{
  int64 duration = 1;
  uint64 files = 2;
};

message remove_root_result
{
  int64 duration = 1;
}

message get_roots_result
{
  int64 duration = 1;
  repeated string roots = 2;
};

service index_service
{
  rpc add_root(root) returns(add_root_result) {}
  rpc remove_root(root) returns(remove_root_result) {}

  rpc get_roots(empty) returns(get_roots_result) {}
  rpc clear_roots(empty) returns(remove_root_result) {}
}
